plugins/PostProcessingPlugin/scripts/MultiExtColorMix.py

```diff
@@ -0,0 +1,623 @@
+#     If 'Extruders Share Heater' and 'Extruders Share Nozzle' are not enabled in Cura - the post will exit.
+#     This script REQUIRES that M163 and M164 are enabled in the firmware.
+#     Options are: Constant Mix or Gradient Mix.  A Range of Layers, or Entire File. 'Resume Extruder' and 'Purge'.
+#     Additional instances of the Post Processor can be run on different layers and with different mix ratios.
+#     This Post Processor is designed for use with 'single extruder' prints.  You should disable all extruders
+#        except for Extruder#1.  If Cura inserts tool changes in the gcode they will interfere with the
+#        'Virtual' mixing extruder. Prime Tower should be disabled.
+#        If you choose to Purge and if retractions are required before moving to park position, then the Extruder #1 retract distance is always used for those retractions.
+#        Without a 'Purge' the tool changes this Post applies are done 'On-the-Fly' and printing simply continues after any tool change.  If you select to insert a Purge, the tool head will move to the park position and purge itself and then continue.  This may leave a tit on a print.  If at the end of a Mix you want to switch to a single extrudere you should probably add a purge or it may be a bit before the color straightens out.
+#     If you end the mix at a mid-print layer; you can continue with the final mix or switch to another extruder.
+#   The sum of the Start Percent for the extruders enabled in this post must equal 100.
+#   The sum of the End Percent (when Gradient is used) for the extruders enabled in this post must also equal 100.
+#   Ex: If you are using two extruders in the mix and one starts at 66 then the other must start at 34.
+#   Ex: If you are using three extruders and the first starts with 35, and 2nd with 10 then E3 must start with 55.
+#   The same thing applies to the End numbers when you use a gradient mix.
+#   If the Percentage Sums do not add up then the post will exit with a message.
+#     The virtual extruder is always Extruder_count+1 (either T2 or T3 in the gcode)
+#     This Post can also be used to simply set an extruder for a print.
+#   May be used with either Absolute or Relative extrusion.
+#   If 'One-at-a-Time' printing and only Extruder #1 is enabled, ALL extruders will be available post-process.
+
+from ..Script import Script
+from UM.Application import Application
+from UM.Message import Message
+from cura.CuraApplication import CuraApplication
+import re
+
+class MultiExtColorMix(Script):
+    def __init__(self):
+        super().__init__()
+    # Enable the third extruder setting if there is one---------------------------------
+    def initialize(self) -> None:
+        super().initialize()
+            
+        ext_cnt = Application.getInstance().getGlobalContainerStack().getProperty("machine_extruder_count", "value")
+        if ext_cnt > 2:
```

Comment:
Cura can have up to 8 extruders, what happens when this script is enabled, and a 4th or even 5th extruder is enabled?

Reply:
Added a message and exit (at line 253) if Extruder Count > 3.